File: src/main/java/cz/solutions/cockroach/EsppReport.java
package cz.solutions.cockroach;

import lombok.Value;

import java.util.List;
import java.util.Map;

import static cz.solutions.cockroach.FormatingHelper.formatDouble;

@Value
public class EsppReport {
    List<PrintableEspp> printableEsppList;
    double profitCroneValue;
    double profitDolarValue;
    int totalEsppAmount;

    public Map<String,?> asMap(){
        return Map.of(
                "esppList", printableEsppList,
                "profitCroneValue", formatDouble(profitCroneValue),
                "profitDolarValue", formatDouble(profitDolarValue),
                "totalAmount", totalEsppAmount
        );
    }
}

File: src/main/kotlin/cz/solutions/cockroach/EsppReport.kt
package cz.solutions.cockroach

data class EsppReport(
    val printableEsppList: List<PrintableEspp>,
    val profitCroneValue: Double,
    val profitDolarValue: Double,
    val totalEsppAmount: Double,
    val taxableProfitCroneValue: Double
) {
    fun asMap(): Map<String, Any> {
        return mapOf(
            "esppList" to printableEsppList,
            "profitCroneValue" to FormatingHelper.formatDouble(profitCroneValue),
            "profitDolarValue" to FormatingHelper.formatDouble(profitDolarValue),
            "totalAmount" to totalEsppAmount,
            "taxableProfitCroneValue" to FormatingHelper.formatDouble(taxableProfitCroneValue)
        )
    }
}